pkg/jwt/decode_test.go

```diff
@@ -3,74 +3,111 @@ package jwt
-			ExpectVal:    `{"payload":{"iat":"Wed, 17 Jan 2018 20:30:22 EST","name":"John Doe","sub":"1234567890"}}`,
+			EnvTimeZone:  "UTC",
+			ExpectVal:    `{"payload":{"iat":"Thu, 18 Jan 2018 01:30:22 UTC","name":"John Doe","sub":"1234567890"}}`,
 			ExpectErr:    true,
 			ExpectErrVal: "header section is not base64url encoded",
 		},
 		{
-			Name:         "Error Fully Invalid JWT #1",
+			Name:         "When input payload is valid but header is not valid json expecting return payload result and error message",
+			InputVal:     `aGVsbG8gd29ybGQ.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.3FMRR0q2EpYL7JAuaYFxFz9mtjNoveYU8HtievNWsXw`,
+			EnvTimeZone:  "UTC",
+			ExpectVal:    `{"payload":{"iat":"Thu, 18 Jan 2018 01:30:22 UTC","name":"John Doe","sub":"1234567890"}}`,
+			ExpectErr:    true,
+			ExpectErrVal: "header section is not valid JSON",
+		},
+		{
+			Name:         "When both input payload and header are not in base64url encoded expecting return empty result and error message",
+			InputVal:     `eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lPz8/IiwiaWF0IjoxNTE2MjM5MDIyfQ.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lPz8/IiwiaWF0IjoxNTE2MjM5MDIyfQ.3FMRR0q2EpYL7JAuaYFxFz9mtjNoveYU8HtievNWsXw`,
+			EnvTimeZone:  "UTC",
+			ExpectVal:    `{}`,
+			ExpectErr:    true,
+			ExpectErrVal: "header section is not base64url encoded; payload section is not base64url encoded",
+		},
+		{
+			Name:         "When both input payload and header are not valid json expecting return empty result and error message",
+			InputVal:     `aGVsbG8gd29ybGQ.aGVsbG8gd29ybGQ.3FMRR0q2EpYL7JAuaYFxFz9mtjNoveYU8HtievNWsXw`,
+			EnvTimeZone:  "UTC",
+			ExpectVal:    `{}`,
+			ExpectErr:    true,
+			ExpectErrVal: "header section is not valid JSON; payload section is not valid JSON",
+		},
+		{
+			Name:         "When input is not valid in the format of a JWT expecting return empty result and error message",
 			InputVal:     `jwt`, // wildly invalid input
 			ExpectVal:    `{}`,
 			ExpectErr:    true,
 			ExpectErrVal: `supplied string is not in the format of a JWT`,
 		},
 	}
 
-	for _, test := range table {
-		t.Run(test.Name,
-			func(t *testing.T) {
-				decoded, err := DecodeJwt(test.InputVal)
-				if test.ExpectErr {
-					if err == nil {
-						fmt.Println("Test expected error but recieved none")
-						t.Fail()
-					} else if err.Error() != test.ExpectErrVal {
-						fmt.Printf("Test failed. Expected Error:\n%s\nGot:\n%s\n", test.ExpectErrVal, err.Error())
-						t.Fail()
-					}
-				} else {
-					if err != nil {
-						t.Fail()
-					}
+	for _, test := range testCases {
+		t.Run(test.Name, func(t *testing.T) {
+			//setting timezone so test results won't be different between different local timezone
+			os.Setenv("TZ", test.EnvTimeZone)
```

Comment:
**suggestion:** collect the current TZ setting and either unset TZ or reset it to the original value with a deferred func